qiskit/dagcircuit/dagdependency.py

```diff
@@ -398,26 +400,34 @@ def _key(x):
 
         return iter(rx.lexicographical_topological_sort(self._multi_graph, key=_key))
 
-    def add_op_node(self, operation, qargs, cargs):
-        """Add a DAGDepNode to the graph and update the edges.
+    def create_op_node(self, operation, qargs, cargs):
+        """Creates a DAGDepNode to the graph and update the edges.
```

Comment:
Exactly, the existing public `add_op_node` function both creates a new `DAGDepNode` (internally calling `DAGDepNode.__init__`) **and** also updates the edges. For contracting a block of nodes using rustworkx `_multi_graph.contract_nodes`, we need to create a new `DAGDepNode` without updating the edges. To avoid code duplication, the node creation part was moved to a separate function. Per your suggestion, I have renamed `create_op_node` to `_create_op_node` (now it's considered to be private, correct?).